File: apps/web/src/features/items/types/item.ts
export interface Item {
  id: string;
  workspaceId: string;
  containerId: string;
  name: string;
  quantity: number;
  category?: string | null;
  source: 'MANUAL' | 'AI_CONFIRMED';
  isVerified: boolean;
  isArchived: boolean;
  createdAt: string;
  updatedAt: string;
}

export interface CreateItemPayload {
  name: string;
  quantity?: number;
  category?: string;
}

export interface UpdateItemPayload {
  name?: string;
  quantity?: number;
  category?: string | null;
}

File: apps/api/WherezIt.Domain/Entities/Item.cs
namespace WherezIt.Domain.Entities;

public class Item
{
    public Guid Id { get; set; }
    public Guid WorkspaceId { get; set; }
    public Guid ContainerId { get; set; }
    public string Name { get; set; } = null!;
    public int Quantity { get; set; } = 1;
    public string? Category { get; set; }
    public string Source { get; set; } = null!;
    public bool IsVerified { get; set; }
    public bool IsArchived { get; set; }
    public DateTimeOffset CreatedAt { get; set; }
    public DateTimeOffset UpdatedAt { get; set; }

    public Workspace Workspace { get; set; } = null!;
    public Container Container { get; set; } = null!;
}
